### PR TITLE
using geopandas and shapely package to calculate whether there are intersections between bbox and sentinel SAFE file.

### PR DESCRIPTION
**Pull request by ranneylxr**
Using geopandas and shapely package to calculate whether there are intersections between bbox and sentinel SAFE file.
And remove "import geopandas and shapely" part into generate_geopolygon function.

### CODE REVIEW

**Comment by piyushrpt** [comment]
Just noting that introduces a new dependency on geopandas. The same can be done with `polygon1.intersects(polygon2)` in shapely?

**Review by yunjunz — CHANGES_REQUESTED** [review]
Thank you @ranneylxr for improving this. I would second @piyushrpt's suggestion on removing `geopandas` as it seems like just using shapely should be enough. [Here](https://gis.stackexchange.com/questions/90055/finding-if-two-polygons-intersect-in-python) is another similar topic on StackExchange. Otherwise, adding a new dependency would result in quite some changes on the installation side.

**Comment by ranneylxr** [comment]
Hi @piyushrpt and @yunjunz . Thank you for your valuable suggestions. Now the script only uses shapely package. Please check.

File: contrib/stack/topsStack/stackSentinel.py [hunk 1]
```diff
@@ -196,6 +196,27 @@ def cmdLineParse(iargs = None):
...
+    p4 = Point(bbox[3][0], bbox[3][1])
+
+    np1 = (p1.coords.xy[0][0], p1.coords.xy[1][0])
+    np2 = (p2.coords.xy[0][0], p2.coords.xy[1][0])
+    np3 = (p3.coords.xy[0][0], p3.coords.xy[1][0])
+    np4 = (p4.coords.xy[0][0], p4.coords.xy[1][0])
+
+    bb_polygon = Polygon([np1, np2, np3, np4])
+
+    return bb_polygon
```
Comment (rtburns-jpl) [inline contrib/stack/topsStack/stackSentinel.py:218]: I think using a list comprehension makes this a bit simpler
```suggestion
    points = [Point(bbox[i][0], bbox[i][1]) for i in range(4)]
    return Polygon([(p.coords.xy[0][0], p.coords.xy[1][0]) for p in points])
```

**Comment by yunjunz** [comment]
Since `matplotlib.path.Path` is not used anymore, could you please also remove [stackSentinel.py#L14-L18](https://github.com/isce-framework/isce2/blob/main/contrib/stack/topsStack/stackSentinel.py#L14-L18)?

**Review by yunjunz — APPROVED** [review]
Looks good and clean. Thank you @ranneylxr!

**Review by hfattahi — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**